test/select.jl

```diff
@@ -0,0 +1,838 @@
+    @test d.d !== df.d
+    @test df == df2
+
+    df2 = copy(df)
+    d = select(df, Not(r"[aec]"))
+    @test d isa DataFrame
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d == df[:, r"[bd]"]
+    @test d.b !== df.b
+    @test d.d !== df.d
+    @test df == df2
+
+    d = select(df, Not([2, 5, 3]))
+    @test d isa DataFrame
+    @test names(d) == [:a, :d]
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d == df[:, [:a, :d]]
+    @test df == df2
+
+    d = select(df, Not(2:3))
+    @test d isa DataFrame
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d.e !== df.e
+    @test df == df2
+
+    d = select(df, Not([false, true, true, false, false]))
+    @test d isa DataFrame
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d.e !== df.e
+    @test df == df2
+
+    d = select(df, Not(1))
+    @test d isa DataFrame
+    @test d == DataFrame(b=2,c=3,d=4,e=5)
+    @test d.b !== df.b
+    @test d.b == df.b
+    @test df == df2
+
+    d = select(df, Not([:a, :e, :c]), copycols=false)
+    @test d isa SubDataFrame
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d.b === df.b
+    @test d.d === df.d
+    @test df == df2
+
+    d = select(df, Not(r"[aec]"), copycols=false)
+    @test d isa SubDataFrame
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d == df[:, r"[bd]"]
+    @test d.b === df.b
+    @test d.d === df.d
+    @test df == df2
+
+    d = select(df, Not([2, 5, 3]), copycols=false)
+    @test d isa SubDataFrame
+    @test names(d) == [:a, :d]
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d == df[:, [:a, :d]]
+    @test df == df2
+
+    d = select(df, Not(2:3), copycols=false)
+    @test d isa SubDataFrame
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d.e === df.e
+    @test df == df2
+
+    d = select(df, Not([false, true, true, false, false]), copycols=false)
+    @test d isa SubDataFrame
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d.e === df.e
+    @test df == df2
+
+    d = select(df, Not(1), copycols=false)
+    @test d isa SubDataFrame
+    @test d == DataFrame(b=2,c=3,d=4,e=5)
+    @test d.b === df.b
+    @test df == df2
+end
+
+@testset "select!" begin
+    df = DataFrame(a=1, b=2, c=3, d=4, e=5)
+    @test_throws ArgumentError select!(df, 0)
+    @test_throws ArgumentError select!(df, 6)
+    @test_throws ArgumentError select!(df, [1, 1])
+    @test_throws ArgumentError select!(df, :f)
+    @test_throws BoundsError select!(df, [true, false])
+
+    @test_throws MethodError select!(view(df, :, :), 1:2)
+
+    d = copy(df, copycols=false)
+    @test select!(d, 1:0) == DataFrame()
+    @test select!(d, Not(r"")) == DataFrame()
+
+    d = copy(df, copycols=false)
+    select!(d, [:a, :e, :c])
+    @test names(d) == [:a, :e, :c]
```

Comment:
Same remark about testing the values instead of just names.

Reply:
here we check all the values anyway for `===` so nothing needs to be changed in tests.